specs/AppNotifications/AppNotificationBuider/GroupAndSubgroup-spec.md

```diff
@@ -0,0 +1,182 @@
+# Groups and Subgroups
+
+Using Groups and Subgroups, developers gain extra control on how they organize text and images in their AppNotifications as shown in the bottom part of the AppNotification in the image below where the text is split into two columns, one of which is right aligned and some of the text is bolded.
+
+![AppNotification Group and Subgroup Example](toast-content-columns.png)
+
+The kind of fancy formatting shown in the above image can be achieved with the following xml, but it cannot be achieved using the AppNotificationBuilder at the moment.
+
+```xml
+<toast>
+    <visual>
+        <binding template='ToastGeneric'>
+            <text>Reminder: Windows Launch Party</text>
+            <text>4:00 PM, 10/21/2021</text>
+            <group>
+                <subgroup>
+                    <text hint-style='base'>52 attendees</text>
+                    <text hint-style='captionSubtle'>23 minute drive</text>
+                </subgroup>
+                <subgroup>
+                    <text hint-style='captionSubtle' hint-align='right'>1 Microsoft Way</text>
+                    <text hint-style='captionSubtle' hint-align='right'>Bellevue, WA 98008</text>
+                </subgroup>
+            </group>
+        </binding>
+    </visual>
+</toast>
+```
+
+To support the Groups and Subgroups in the AppNotificationBuilder will require two new runtimeclasses: one for Groups and one for Subgroups.
+
+The new AppNotificationGroup runtimeclass will simply be a container of Subgroups.
+
+```idl
+[contract(AppNotificationBuilderContract, 2)]
+runtimeclass AppNotificationGroup
+{
+    AppNotificationGroup();
+
+    Windows.Foundation.Collections.IVector<AppNotificationSubgroup> subgroups;
+
+    AppNotificationGroup AddSubgroup(AppNotificationSubgroup subgroup);
```

Comment:
How many groups and subgroups can be added to an app notification?

Reply:
Don't know. The doc I was looking at doesn't specify but from my experiement with Notifications Visualizer there doesn't seem to be a limit. 

The only thing is that visually  more than two subgroups doesn't seem to make sense.